core/network/src/test/mod.rs

```diff
@@ -591,6 +591,9 @@ pub trait TestNetFactory: Sized {
 		// Return `NotReady` if there's a mismatch in the highest block number.
 		let mut highest = None;
 		for peer in self.peers().iter() {
+			if peer.is_major_syncing() || peer.network.num_queued_blocks() != 0 {
+				return Async::NotReady
```

Comment:
Is that really okay? Are we sure that something wakes up the task to continue?

Reply:
Queue events wake up the task I believe.